src/impl_float_maths.rs

```diff
@@ -0,0 +1,157 @@
+//! Element-wise methods for ndarray
+
+use num_traits::Float;
+
+use crate::imp_prelude::*;
+
+macro_rules! boolean_op {
```

Comment:
> What is your justification for this overarching macro that takes multiple function stubs and outputs multiple functions, versus the more common single-function macro like ndarray's impl_binary_op!?

I'm prefer to use keyword syntax (`fn`, `impl`) in the implementation. In my opinion, function-like calling seems to be a brief-function of the expressions or statements. Since there are many docstrings displayed, perhaps using `fn` keyword is a better syntax (docstring is a kind of attribute macros).

Reply:
If I'm understanding correctly, you prefer this type of macros that *don't* behave like functions? (and indeed this one does not behave like a function).

I'm pretty sure you can easily write docstrings with either approach, though. You would just:
```rust
/// The ceiling function
impl_float_op!(ceil); 
```

Anyway don't change this on my account, we'll have to see what the maintainers say.

Reply:
The current macros look smart, so whatever you like @KmolYuan. If you want you can rename them with plural, "binary_ops" since they produce multiple methods (maybe even make it "binary_methods"?)